cartography/rules/runners.py

```diff
@@ -0,0 +1,264 @@
+"""
+Framework and Fact execution logic for Cartography rules.
+"""
+
+import json
+from dataclasses import asdict
+
+from neo4j import Driver
+from neo4j import GraphDatabase
+
+from cartography.rules.data.frameworks import FRAMEWORKS
+from cartography.rules.formatters import _generate_neo4j_browser_url
+from cartography.rules.spec.model import Fact
+from cartography.rules.spec.model import Framework
+from cartography.rules.spec.model import Requirement
+from cartography.rules.spec.output import FactResult
+from cartography.rules.spec.output import FrameworkResult
+
+
+def _run_fact(
+    fact: Fact,
+    requirement: Requirement,
+    framework: Framework,
+    driver: Driver,
+    database: str,
+    fact_counter: int,
+    total_facts: int,
+    output_format: str,
+    neo4j_uri: str,
+):
+    """Execute a single fact and return the result."""
+    if output_format == "text":
+        print(f"\n\033[1mFact {fact_counter}/{total_facts}: {fact.name}\033[0m")
+        print(f"  \033[36m{'Framework:':<12}\033[0m {framework.name}")
+        # Display requirement with optional clickable link
+        if requirement.requirement_url:
+            print(
+                f"  \033[36m{'Requirement:':<12}\033[0m \033]8;;{requirement.requirement_url}\033\\{requirement.id}\033]8;;\033\\"
+            )
+        else:
+            print(f"  \033[36m{'Requirement:':<12}\033[0m {requirement.id}")
+        print(f"  \033[36m{'Fact ID:':<12}\033[0m {fact.id}")
+        print(f"  \033[36m{'Description:':<12}\033[0m {fact.description}")
+        print(f"  \033[36m{'Provider:':<12}\033[0m {fact.provider.value}")
+
+        # Generate and display clickable Neo4j Browser URL
+        browser_url = _generate_neo4j_browser_url(neo4j_uri, fact.cypher_visual_query)
+        print(
+            f"  \033[36m{'Neo4j Query:':<12}\033[0m \033]8;;{browser_url}\033\\Click to run visual query\033]8;;\033\\"
+        )
+
+    with driver.session(database=database) as session:
+        result = session.run(fact.cypher_query)
+        findings = [dict(record) for record in result]
+        finding_count = len(findings)
```

Comment:
Could we use `Driver.execute_query()` for everything in here the rules module? Would be good to have transactions on everything